eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```diff
@@ -29,35 +29,92 @@ jobs:
           OSVmImage: 'macOS-10.15'
           vcpkg.deps: 'curl[ssl]'
           VCPKG_DEFAULT_TRIPLET: 'x64-osx'
-          build.args: ' -DTRANSPORT_CURL=ON  -DAZ_PLATFORM_IMPL=POSIX'
-
+          build.args: ' -DTRANSPORT_CURL=ON -DAZ_PLATFORM_IMPL=POSIX'
+        Win_x64_with_iot_samples:
+          OSVmImage: 'windows-2019'
+          vcpkg.deps: 'curl[winssl] cmocka paho-mqtt'
+          VCPKG_DEFAULT_TRIPLET: 'x64-windows-static'
+          CMAKE_GENERATOR: 'Visual Studio 16 2019'
+          CMAKE_GENERATOR_PLATFORM: x64
+          build.args: ' -DTRANSPORT_PAHO=ON'
+          test_type: 'iot'
+        Linux_x64_with_iot_samples:
+          OSVmImage: 'ubuntu-18.04'
+          vcpkg.deps: 'curl[ssl] cmocka paho-mqtt'
+          VCPKG_DEFAULT_TRIPLET: 'x64-linux'
+          build.args: ' -DTRANSPORT_PAHO=ON'
+          test_type: 'iot'
     pool:
       vmImage: $(OSVmImage)
 
     steps:
       - template: /eng/pipelines/templates/steps/vcpkg.yml
         parameters:
           DependenciesVariableName: vcpkg.deps
+          
+      - template: /eng/common/TestResources/deploy-test-resources.yml
+        parameters:
+          ServiceDirectory: '.'
+
+      - pwsh: |
+          # Generate certificate 
+          cd $SOURCES_DIR\sdk\samples\iot\
+          openssl ecparam -out device_ec_key.pem -name prime256v1 -genkey
+          openssl req -new -days 365 -nodes -x509 -key device_ec_key.pem -out device_ec_cert.pem -config x509_config.cfg -subj "/CN=$DEVICE_ID"
+
+          Get-Content device_ec_cert.pem, device_ec_key.pem | Set-Content device_cert_store.pem
+          openssl x509 -noout -fingerprint -in device_ec_cert.pem | % {$_.replace(":", "")} | % {$_.replace("SHA1 Fingerprint=", "")} | Tee-Object fingerprint.txt
+          $fingerprint = Get-Content -Path .\fingerprint.txt
+          
+          # Pass fingerprint to IoTHub 
+          Add-AzIotHubDevice `
+            -ResourceGroupName $ResourceGroupName `
+            -IotHubName $IOTHUB_NAME `
+            -DeviceId $DEVICE_ID `
+            -AuthMethod "x509_thumbprint" `
+            -PrimaryThumbprint $fingerprint`
+            -SecondaryThumbprint $fingerprint
+            
+          # Download Baltimore Cert
+          curl https://cacerts.digicert.com/BaltimoreCyberTrustRoot.crt.pem > $SOURCES_DIR\sdk\samples\iot\BaltimoreCyberTrustRoot.crt.pem
```

Comment:
Given you are in a powershell context I believe curl is an alia for Invoke-WebRequest. We generally advice you to use the powershell commandline by name instead of by alias as it is less surprising when running on different platforms. (i.e on one platform it might use the curl ext and the other it might use Invoke-WebRequest).

Reply:
curl is it's own tool, I noticed that curl was installed a part of the vcpkg dependencies, so I figured it was completely kosher to use it since it's guaranteed installed. I was also not familiar with Invoke-WebRequest. curl is also available on a monumental number of platforms.

Reply:
Ok I just double checked at https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/InitialSessionState.cs and curl is not a default alias as I thought so this should be fine to use if you are sure it is available in all your environments. However you can also use Invoke-WebRequest as well as that is a Powershell feature and will also be available everywhere PS is.